### ISSUE TITLE
Crash when closing scenes after interacting with the animation player

### ISSUE THREAD

**Issue by nikitalita** [godotengine/godot#86602]
### Tested versions

Current master (13a0d6e9b253654f5cc2a44f3d0b3cae10440443)

### System information

MacOS - Godot v4.3-13a0d6e9

### Issue description

Getting crashes with EA_BAD_ACCESS when closing scenes after interacting with the animation player under certain conditions.
The scenario is: opening the scene, interacting with the animation player, highlighting something else, and then closing it.
It can crash with both non-inherited and inherited scenes, but is more easily reproduced with inherited scenes from newly-imported scenes (i.e. glb, dae, etc.)

I thought it might have to do with https://github.com/godotengine/godot/pull/57205#pullrequestreview-1798728003, but changing that line to check for `!obj` first doesn't help.

Backtrace:
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.dev.custom_build (13a0d6e9b253654f5cc2a44f3d0b3cae10440443)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x00000001885bda24 _sigtramp + 56
[2] 2   ???                                 0xffff800188545b64 0x0 + 18446603342803393380
[3] 3   libc++abi.dylib                     0x0000000188545b64 __dynamic_cast + 132
[4] VariantObjectClassChecker<Node*>::check(Variant const&) (in godot.macos.editor.dev.arm64) (binder_common.h:230)
[5] VariantCasterAndValidate<Node*>::cast(Variant const**, unsigned int, Callable::CallError&) (in godot.macos.editor.dev.arm64) (binder_common.h:255)
[6] void call_with_variant_args_helper<GroupsEditor, Node*, 0ul>(GroupsEditor*, void (GroupsEditor::*)(Node*), Variant const**, Callable::CallError&, IndexSequence<0ul>) (in godot.macos.editor.dev.arm64) (binder_common.h:303)
[7] void call_with_variant_args<GroupsEditor, Node*>(GroupsEditor*, void (GroupsEditor::*)(Node*), Variant const**, int, Callable::CallError&) (in godot.macos.editor.dev.arm64) (binder_common.h:418)
[8] CallableCustomMethodPointer<GroupsEditor, Node*>::call(Variant const**, int, Variant&, Callable::CallError&) const (in godot.macos.editor.dev.arm64) (callable_method_pointer.h:99)
[9] Callable::callp(Variant const**, int, Variant&, Callable::CallError&) const (in godot.macos.editor.dev.arm64) (callable.cpp:58)
[10] CallQueue::_call_function(Callable const&, Variant const*, int, bool) (in godot.macos.editor.dev.arm64) (message_queue.cpp:222)
[11] CallQueue::flush() (in godot.macos.editor.dev.arm64) (message_queue.cpp:327)
[12] SceneTree::physics_process(double) (in godot.macos.editor.dev.arm64) (scene_tree.cpp:471)
[13] Main::iteration() (in godot.macos.editor.dev.arm64) (main.cpp:3750)
[14] OS_MacOS::run() (in godot.macos.editor.dev.arm64) (os_macos.mm:766)
[15] main (in godot.macos.editor.dev.arm64) (godot_main_macos.mm:77)
[16] 16  dyld                                0x000000018820d0e0 start + 2360
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

Double click on AnimatedMorphCube.glb, then click "Reimport".
Create a new inherited scene from AnimatedMorphCube.glb
In the Scene tab, Click on `Animation Player`
Then, click on `AnimationMorphCube`
Then, close the scene tab. It should crash with a back trace.

### Minimal reproduction project (MRP)

[another_test_project.zip](https://github.com/godotengine/godot/files/13792066/another_test_project.zip)

**Comment by nikitalita** [comment]
I should note this is most easily reproduced on Debug builds; I am able to get it to crash on Release builds, but only after repeating the steps a number of times.

Backtrace for the release build:
```

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.dev.custom_build (13a0d6e9b253654f5cc2a44f3d0b3cae10440443)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x00000001885bda24 _sigtramp + 56
[2] 2   libc++abi.dylib                     0x0000000188545b64 __dynamic_cast + 132
[3] 3   libc++abi.dylib                     0x0000000188545b64 __dynamic_cast + 132
[4] CallQueue::_call_function(Callable const&, Variant const*, int, bool) (in godot.macos.editor.arm64) + 336
[5] CallQueue::flush() (in godot.macos.editor.arm64) + 348
[6] SceneTree::process(double) (in godot.macos.editor.arm64) + 220
[7] Main::iteration() (in godot.macos.editor.arm64) + 984
[8] OS_MacOS::run() (in godot.macos.editor.arm64) + 116
[9] main (in godot.macos.editor.arm64) + 304
[10] 10  dyld                                0x000000018820d0e0 start + 2360
-- END OF BACKTRACE --
================================================================
```

**Comment by Rindbee** [comment]
Seems to be a duplicate of #86427.

**Comment by nikitalita** [comment]
I should note, however, that since I was getting this on the release builds, and the PR only changes the function for debug builds, this may not completely fix the issue
